Automattic/pocket-casts-android
# [Paywall Experiments] Add Paywall features cards

## Description
- Adds the feature cards for paywall `FEATURE CAROUSEL` type
- See designs: SEQe6JBDCeegcpa9Bfg5Ww-fi-3_756
- Slack discussion: p1723644311582689-slack-C07G1MH890E
- After I connect these cards to the new layout I will be able to tests landscape cases and accessibility

In a later PR I will add these cards to the new layout added here: https://github.com/Automattic/pocket-casts-android/pull/2635

Fixes #2629

## Testing Instructions
- See the `modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/upgrade/paywallfeatures/PaywallCards.kt` file and open the Jetpack Compose preview

## Screenshots or Screencast 

![image](https://github.com/user-attachments/assets/e8c7ea16-e0a4-4d24-a57f-bc0fee402af2)


## Checklist
 
#### I have tested any UI changes...
